egs/wsj/s5/steps/nnet3/multilingual/allocate_multilingual_examples.py

```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/env python
+
+# This script generates egs.Archive.scp and ranges.* used for generating egs.Archive.scp 
+# for multilingual setup.
+# Also this script generates outputs.*.scp and weight.*.scp, where each line 
+# corresponds to language-id and weight for the same example in egs.*.scp. 
+# weight.*.scp used to scale the output's posterior during training.
+# ranges.*.scp is generated w.r.t frequency distribution of remaining examples 
+# in each language. 
+#
+# You call this script as (e.g.)
+#
+# allocate_multilingual_examples.py [opts] num-of-languages example-scp-lists multilingual-egs-dir
+#
+# allocate_multilingual_examples.py --num-jobs 10 --samples-per-iter 10000 --minibatch-size 512
+# --lang2weight exp/multi/lang2weight 2 "exp/lang1/egs.scp exp/lang2/egs.scp" 
+# exp/multi/egs
+#
+# This script outputs specific ranges.* files to the temp directory (exp/multi/egs/temp)
+# that will enable you to creat egs.*.scp files for multilingual training.
+# exp/multi/egs/temp/ranges.* contains something like the following:
+# e.g.
+# lang1 0 0 256
+# lang2 1 256 256
+#
+# where each line can be interpreted as follows:
+# <source-language> <local-scp-line> <num-examples>
+#
+# note that <local-scp-line> is the zero-based line number in egs.scp for 
+# that language.
+# num-examples is multiple of actual minibatch-size.
+#
+#
+# egs.1.scp is generated using ranges.1.scp as following:
+# "num_examples" consecutive examples starting from line "local-scp-line" from 
+# egs.scp file for language "source-lang" is copied to egs.1.scp.
+#
+#
+
+from __future__ import print_function
+import re, os, argparse, sys, math, warnings, random, io, imp
+import numpy as np
+
```

Comment:
this file should not have been included in your patch, I think something went wrong resolving the merge.